public/app/features/trails/MetricSelect/util.ts

```diff
@@ -10,8 +10,20 @@ export function deriveSearchTermsFromInput(whiteSpaceSeparatedTerms?: string) {
   );
 }
 
-export function createJSRegExpFromSearchTerms(searchQuery?: string) {
-  const searchParts = deriveSearchTermsFromInput(searchQuery).map((part) => `(?=(.*${part.toLowerCase()}.*))`);
+export function createJSRegExpFromSearchTerms(searchQuery?: string, prefix?: boolean) {
+  let searchParts: string[] = [];
+
+  // build a regex for either
+  // the search metrics input(contains)
+  // or
+  // the metricsPrefix(starts with)
+  if (prefix) {
+    // there will only be one prefix selected at a time
+    searchParts = [`(?=^${searchQuery}.*)`];
+  } else {
+    // there may be multiple search terms
+    searchParts = deriveSearchTermsFromInput(searchQuery).map((part) => `(?=(.*${part.toLowerCase()}.*))`);
+  }
```

Comment:
So it looks like the strategy is to either be a prefix or a search query of multiple terms, but never both at the same time?

Reply:
Maybe in a future iterations these could be separate functions?
Or if we want a single regex to consider both, a single function with searchQuery and prefix as a string?

(No need to change this PR)

Reply:
Yep! I thought they would be both at the same time at first, but then I realized the node groups needed to be re-generated after the metric search input. So, I separated them. 

Since they are separate, the metric prefix probably doesn't even need a function, or the additional `(?=<regex>)`.

I can take this out!